test/unit/ready.js

```diff
@@ -67,4 +70,20 @@ QUnit.module( "ready" );
 			"Argument passed to fn in jQuery(document).ready( fn ) should be jQuery" );
 	} );
 
+	QUnit.test( "Promise.resolve(jQuery.ready)", function( assert ) {
+		assert.expect( 2 );
+
+		var beforeReady = assert.async();
+		var afterReady = assert.async();
```

Comment:
Betterr:
```js
var done = assert.async( 2 );
```
and then just call `done()` in both callbacks.

That is, if we have new enough QUnit.